### PR TITLE
Add cloud image pipeline

### PR DESCRIPTION
**Pull request by jlebon**
Add another Jenkins pipeline `Jenkinsfile.cloud` for generating cloud
images. I also renamed the treecompose one to `Jenkinsfile.treecompose`.
The kickstart was lightly adapted from `fedora-atomic`.

### CODE REVIEW

**Comment by jlebon** [comment]
/hold

Still sanity checking results.

**Comment by jlebon** [comment]
OK, image works!
/hold cancel

File: cloud.ks [hunk 1]
```diff
@@ -0,0 +1,128 @@
+text # don't use cmdline -- https://github.com/rhinstaller/anaconda/issues/931
```
Comment (peterbaouoft) [inline cloud.ks]: question :), so I am guessing when running the pipeline job, this ks file will be visible on the node too? What is the reasoning behind that? Did jenkins use the checkedout repo location as its working directory by any chance? (/me asking with little knowledge here, so forgive my ignorance :p)

File: rhcos.tdl [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<template>
+    <name>rhcos</name>
+    <os>
+        <name>RHEL-7</name>
+        <version>0</version>
+        <arch>x86_64</arch>
+        <install type='url'>
+            <url>https://dl.fedoraproject.org/pub/fedora/linux/releases/28/Everything/x86_64/os/</url>
```
Comment (peterbaouoft) [inline rhcos.tdl]: Minor: I am guessing this url will be swapped, once we publish the composed ostree repo(created via the other pipeline) to a similar url?

Comment (peterbaouoft) [reply rhcos.tdl]: Ah, NVM, I think I see, I was confused for a while, I am guessing 
`sh "sed -i 's,\\(<url>\\).*\\(<\\/url\\),\\1${env.INSTALLER_TREE_URL}\\2,' rhcos.tdl"` does the url swap alrdy?

File: cloud.ks (hunk 1 above)
Comment (jlebon) [reply cloud.ks]: You're exactly right! Jenkins will checkout the code into the current directory.

File: rhcos.tdl (hunk 2 above)
Comment (jlebon) [reply rhcos.tdl]: Yes, that's correct.

**Comment by jlebon** [comment]
The kickstart will most certainly need modifications. My initial goal was to just get it in and then iterate on it -- I didn't actually go through it with a fine-tooth comb yet.

File: Jenkinsfile.cloud [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                    -e 'ssh -i ${env.KEY_FILE} \
+                            -o UserKnownHostsFile=/dev/null \
+                            -o StrictHostKeyChecking=no' \
+                    ${env.ARTIFACT_SERVER}:${output}/ ${output}
+            """
+        }
+    }
+
+    stage("Prepare Configs") {
+        sh "sed -i 's,\\(<url>\\).*\\(<\\/url\\),\\1${env.INSTALLER_TREE_URL}\\2,' rhcos.tdl"
```
Comment (miabbott) [inline Jenkinsfile.cloud]: RIP my brain

**Pull request merged**